frame/clearing-house/src/lib.rs

```diff
@@ -1200,10 +1244,95 @@ pub mod pallet {
 				}
 			}
 
+			// Charge fees
+			let liquidator_fee =
+				Self::full_liquidation_penalty_liquidator_share().saturating_mul_int(fees);
+			let insurance_fee = fees.try_sub(&liquidator_fee)?;
+
 			Positions::<T>::insert(user_id, positions);
 			Collateral::<T>::insert(user_id, collateral);
 
-			Ok(fees)
+			Ok((liquidator_fee, insurance_fee))
+		}
+
+		/// Partially liquidates the user's positions until its account is brought above the PMR.
+		///
+		/// This function does **not** check if the account is below the PMR beforehand.
+		///
+		/// ## Storage modifications
+		///
+		/// - Updates the [`markets`](Markets) of decreased positions (according to changes made by
+		///   [`Self::decrease_position`])
+		/// - Updates reduced [`positions`](Positions) (according to changes made by
+		///   [`Self::decrease_position`])
+		/// - Updates the user's account [`collateral`](Collateral)
+		///
+		/// ## Returns
+		///
+		/// The fees for the liquidator and insurance fund
+		fn partially_liquidate_account(
```

Comment:
Functionality wise this function is solid. I think, in the future, it would be beneficial to abstract parts of this function away into their own private helper functions to help shorten the length and bulkiness of the entire function.

Reply:
For sure; some refactoring is due. I'll try to include that with the more sophisticated checks to be implemented later on